### PR TITLE
405 cors refactor for ie11

### PR DESCRIPTION
**Pull request by briandavidson**
﻿## Summary
Addresses Issue # 405

Please note if fully resolves the issue per the acceptance criteria: Yes

*Describe the pull request here, including any supplemental information needed to understand it.*
This pull request refactors our existing CORS implementation to instead use the CORS package. 

## Impacted Areas of the Site
- /admin/christmas-trees

## This pull request changes...
- [x] This resolves an issue where admin users were not able to login using Eauth

## This pull request is ready to merge when...
- [x] Feature branch is starts with the issue number
- [x] Is connected to its original issue via zenhub 👇
- [x] Tests have been updated 
- [x] All tests are passing and meet coverage, linting, and accessibility requirements. And no security vulnerabilities ⚫️(Circle)
- [x] Server actions captured by logs (manual)
- [x] Documentation / readme.md updated (manual)
- [x] API docs updated if need (manual)
- [x] JSDocs updated (manual)
- [ ] This code has been reviewed by someone other than the original author

### CODE REVIEW

File: server/src/routers/router.es6 [hunk 1]
```diff
@@ -12,18 +15,31 @@ const specialUseRouter = require('./specialuse.es6');
...
+  allowedHeaders: 'accept, content-type',
+  methods: 'GET, PUT, POST, DELETE, OPTIONS, PATCH'
+};
+
 /** Allow any server to check the preflights. */
-router.options('*', middleware.setCorsHeaders, (_req, res) => {
+router.options('*', cors(corsOpts), (_req, res) => {
   res.set('Access-Control-Allow-Headers', 'accept, content-type');
   res.set('Access-Control-Allow-Methods', 'GET, PUT, POST, DELETE, OPTIONS, PATCH');
+  res.set('Cache-Control', 'no-cache');
```
Comment (davemcorwin) [inline server/src/routers/router.es6]: We want the cache header to apply to ALL of the requests, by placing it here, it only applies to the `OPTIONS` requests. I think it should be added at the route, for all requests, along the lines of the sample code I sent you earlier.

Comment (briandavidson) [reply server/src/routers/router.es6]: Should be all set now, thanks David!

**Comment by briandavidson** [comment]
> Also, please remove the middleware method if it is no longer being used.

The middleware method is no longer being used in [router.es6](https://github.com/18F/fs-open-forest-platform/blob/dev/server/src/routers/router.es6) or in any of the tests.

However there are still references to it in some of our mock files such as [this one.](https://github.com/18F/fs-open-forest-platform/blob/3a63216677b2b5e8e0e1ea517b67d4fa163d8dfd/server/src/mocks/login-gov-mocks.es6#L13)

I'm not entirely clear what current dependencies there are on those mock files or if it's within the scope of this task to make changes to them so for now I left the middleware method in place.

@davemcorwin please advise

**Comment by davemcorwin** [comment]
@briandavidson this was approved a few days ago, anything holding up merging it in?

**Comment by briandavidson** [comment]
> @briandavidson this was approved a few days ago, anything holding up merging it in?

@davemcorwin nope, should be good to go. Shall I go ahead and merge it?

**Comment by davemcorwin** [comment]
@briandavidson The approval typically indicates that it is ready to merge, unless there are specific prerequisites that have to be completed.

**Comment by briandavidson** [comment]
@davemcorwin got it. I'll go ahead and merge then.

**Comment by briandavidson** [comment]
@ASprinkle in regard to you still experiencing the issue in production -- I'm wondering if it was perhaps  just a caching issue:

image

Could we try testing this again after clearing your web browser's cache?

**Pull request merged**